docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,54 @@
+# Adding a New Metrics Provider for Dummy Endpoint
+
+To create a provider for the dummy endpoint, follow these steps:
+
+1. Fork the [Keptn repo](https://github.com/keptn/lifecycle-toolkit)
+
+2. **Define the Provider Type:** In the `metrics-operator/controllers/common/providers/common.go` file,
+ define the constant `KeptnDummyProviderType` with the value `"dummy"`.
+
+    ```go
+    const KeptnDummyProviderType = "dummy"
+    ```
+
+3. **Implement the Provider:** Create your own new folder inside the
+[metrics-operator/controllers/common/providers](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/controllers/common/providers)
+ matching the new service name: dummy and a new Go package for the dummy provider.
+  This package should contain
+ a struct that implements the `KeptnSLIProvider` interface.
+  You can follow other existing implementations,
+ such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+ as an example.
+  Ensure that you implement the `EvaluateQuery` function to fetch the metrics accurately.
```

Comment:
This sentence is easy to misunderstand, to implement the entire KeptnSLIProvider interface one would also have to 
write the functions `EvaluateQueryForStep` and `FetchAnalysisValue ` It would be nice to add a file here with dummy function with a panic 

something like 

```
type MyProvider struct{}

func (p *MyProvider)  EvaluateQuery(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) (string, []byte, error){
	//TODO implement me
	panic("implement me")
}
func (p *MyProvider)  EvaluateQueryForStep(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) ([]string, []byte, error){
  //TODO implement me
	panic("implement me")
}
func (p *MyProvider)  FetchAnalysisValue(ctx context.Context, query string, spec metricsapi.Analysis, provider *metricsapi.KeptnMetricsProvider) (string, error){
	//TODO implement me
	panic("implement me")
}
```

Reply:
That's true.
I will update the doc

Reply:
What is also important for the tutorial is to explain when those 3 methods are called and what does Keptn expect to happen, so at least explaining what will be done with the output of each method